### PR TITLE
performance improvements

### PR DESCRIPTION
**Pull request by barisusakli**
cache nconf.get calls
modify middleware.pageView to call next earlier
don't call winston.verbose on every hook see winstonjs/winston#1669
translate header/footer separately and cache results for guests
replace lodash `_.clone(obj)` with `{ ..obj }` where possible

### CODE REVIEW

File: src/controllers/api.js [hunk 1]
```diff
@@ -15,11 +15,17 @@ const languages = require('../languages');
 
 const apiController = module.exports;
 
+const relative_path = nconf.get('relative_path');
+const upload_url = nconf.get('relative_path');
```
Comment (pitaj) [inline src/controllers/api.js:19]: upload_url != relative_path

File: src/controllers/api.js [hunk 2]
```diff
@@ -15,11 +15,17 @@ const languages = require('../languages');
...
+const socketioOrigins = nconf.get('socket.io:origins');
+const websocketAddress = nconf.get('socket.io:address') || '';
+
 apiController.loadConfig = async function (req) {
 	let config = {
-		relative_path: nconf.get('relative_path'),
-		upload_url: nconf.get('upload_url'),
-		assetBaseUrl: `${nconf.get('relative_path')}/assets`,
+		relative_path: relative_path,
+		upload_url: upload_url,
```
Comment (pitaj) [inline src/controllers/api.js:27]: Can just do `{ relative_path, upload_url, ... }`

File: src/controllers/api.js [hunk 3]
```diff
@@ -40,9 +46,9 @@ apiController.loadConfig = async function (req) {
 		disableChat: meta.config.disableChat === 1,
 		disableChatMessageEditing: meta.config.disableChatMessageEditing === 1,
 		maximumChatMessageLength: meta.config.maximumChatMessageLength || 1000,
-		socketioTransports: nconf.get('socket.io:transports') || ['polling', 'websocket'],
-		socketioOrigins: nconf.get('socket.io:origins'),
-		websocketAddress: nconf.get('socket.io:address') || '',
+		socketioTransports: socketioTransports,
```
Comment (pitaj) [inline src/controllers/api.js:49]: Same as above

File: src/middleware/header.js [hunk 4]
```diff
@@ -54,7 +56,7 @@ async function generateHeader(req, res, data) {
 		allowRegistration: registrationType === 'normal',
 		searchEnabled: plugins.hasListeners('filter:search.query'),
 		config: res.locals.config,
-		relative_path: nconf.get('relative_path'),
+		relative_path: relative_path,
```
Comment (pitaj) [inline src/middleware/header.js:59]: Same here

File: src/controllers/api.js (hunk 1 above)
Comment (barisusakli) [reply src/controllers/api.js:19]: ty 👍

File: src/plugins/hooks.js [hunk 5]
```diff
@@ -87,9 +87,9 @@ module.exports = function (Plugins) {
 	Plugins.fireHook = async function (hook, params) {
 		const hookList = Plugins.loadedHooks[hook];
 		const hookType = hook.split(':')[0];
-		if (hook !== 'action:plugins.firehook') {
-			winston.verbose('[plugins/fireHook] ' + hook);
-		}
+		// if (hook !== 'action:plugins.firehook') {
```
Comment (pitaj) [inline src/plugins/hooks.js:90]: Maybe guard this with global.env = debug?

**Review by pitaj — CHANGES_REQUESTED** [review]

File: src/plugins/hooks.js (hunk 5 above)
Comment (barisusakli) [reply src/plugins/hooks.js]: added global.env==='development' check

**Pull request merged**